client/netdb.c

```diff
@@ -293,22 +293,56 @@ static void *refresh_thread(void *arg)
 	while (!g_xdag_sync_on) {
 		sleep(1);
 	}
-	
+
+	xdag_mess("start refresh_thread");
+
 	for (;;) {
 		time_t prev_time = time(0);
 		
 		xdag_mess("try to refresh white-list...");
 		
 		char *resp = http_get(WHITE_URL);
 		if(resp) {
-			pthread_mutex_lock(&g_white_list_mutex);
-			FILE *f = xdag_open_file(DATABASEWHITE, "w");
-			if(f) {
-				fwrite(resp, 1, strlen(resp), f);
-				fclose(f);
+			char buf[0x1000];
```

Comment:
I will accept this change, but please, move validation to separate function.

Reply:
Please find the new update in #249 .